app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt

```diff
@@ -146,34 +157,40 @@ private fun PlannedHikes(hikes: List<SavedHike>?) {
     }
   } else {
     LazyColumn {
-      items(plannedHikes.size, key = { plannedHikes[it].id }) { index ->
-        val hike = plannedHikes[index]
-        HikeCard(
-            title = hike.name,
-            // This generates a random list of elevation data for the hike
-            // with a random number of points and altitude between 0 and 1000
-            elevationData = (0..(0..1000).random()).map { it.toDouble() }.shuffled(),
-            showGraph = false,
-            onClick = {
-              Toast.makeText(
-                      context,
-                      "Hike details not implemented yet for this screen, since the listOfHikeRoutesVM does not support fetching just a single hike for the moment",
-                      Toast.LENGTH_SHORT)
-                  .show()
-            },
-            messageContent = hike.date!!.humanReadablePlannedLabel(LocalContext.current),
-            modifier = Modifier.testTag(SavedHikesScreen.TEST_TAG_SAVED_HIKES_HIKE_CARD),
-            styleProperties =
-                HikeCardStyleProperties(
-                    messageIcon = painterResource(R.drawable.calendar_today),
-                    messageColor = Color(0xFF3B82F6)))
+      items(plannedHikes.size, key = { plannedHikes[it].value.id }) { index ->
+        val hike by plannedHikes[index].collectAsState()
+        if (!hike.elevation.obtained()) {
+          hikesViewModel.retrieveElevationDataFor(hike.id)
+          HikeCard(
+              title = hike.name ?: stringResource(R.string.map_screen_hike_title_default),
+              elevationData = null,
+              onClick = { hikesViewModel.selectHike(hike.id) },
+              messageContent = hike.plannedDate!!.humanReadablePlannedLabel(LocalContext.current),
+              modifier = Modifier.testTag(SavedHikesScreen.TEST_TAG_SAVED_HIKES_HIKE_CARD),
+              styleProperties =
+                  HikeCardStyleProperties(
+                      messageIcon = painterResource(R.drawable.calendar_today),
+                      messageColor = Color(0xFF3B82F6)))
+        } else {
+          HikeCard(
+              title = hike.name ?: stringResource(R.string.map_screen_hike_title_default),
+              elevationData = hike.elevation.getOrThrow(),
+              onClick = { hikesViewModel.selectHike(hike.id) },
+              messageContent = hike.plannedDate!!.humanReadablePlannedLabel(LocalContext.current),
+              modifier = Modifier.testTag(SavedHikesScreen.TEST_TAG_SAVED_HIKES_HIKE_CARD),
+              styleProperties =
+                  HikeCardStyleProperties(
+                      messageIcon = painterResource(R.drawable.calendar_today),
+                      messageColor = Color(0xFF3B82F6),
+                      graphColor = Color(hike.getColor())))
+        }
```

Comment:
Nitpick: Can you refactor this part to not have the same data computed twice in the code (for example `title`)

Reply:
Done 👌🏻